File: app/src/main/java/com/novel/read/widget/dialog/AppraiseDialog.java
package com.novel.read.widget.dialog;

import android.app.Dialog;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.novel.read.R;

public class AppraiseDialog extends Dialog {
    private View view;
    private TextView mTvSure;
    public AppraiseDialog(@NonNull Context context) {
        super(context, R.style.dialog);
        view = LayoutInflater.from(context).inflate(R.layout.dialog_go_appraise, null);
        view.setHasTransientState(true);
        setContentView(view,new LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.MATCH_PARENT,
                LinearLayout.LayoutParams.MATCH_PARENT));
        setCancelable(false);
        initView();

    }

    private void initView() {
        TextView mTvCancel = findViewById(R.id.tv_cancel);
        mTvSure = findViewById(R.id.tv_appraise);
        mTvCancel.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                dismiss();
            }
        });
    }

    public void AppraiseDialog(View.OnClickListener receiveClickListener){
        mTvSure.setOnClickListener(receiveClickListener);
    }

}

File: app/src/main/java/com/novel/read/ui/widget/dialog/AppraiseDialog.kt
package com.novel.read.ui.widget.dialog

import android.content.Context
import android.os.Bundle
import android.view.*
import com.novel.read.R
import com.novel.read.base.BaseDialogFragment
import com.novel.read.databinding.DialogGoAppraiseBinding
import com.novel.read.utils.ext.goShop
import com.novel.read.utils.viewbindingdelegate.viewBinding

class AppraiseDialog : BaseDialogFragment() {
    private val binding by viewBinding(DialogGoAppraiseBinding::bind)

    override fun onStart() {
        super.onStart()
        dialog?.window?.setLayout(
            ViewGroup.LayoutParams.MATCH_PARENT,
            ViewGroup.LayoutParams.WRAP_CONTENT
        )
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.dialog_go_appraise, container)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setStyle(STYLE_NO_TITLE, R.style.dialog_app)
    }

    override fun onFragmentCreated(view: View, savedInstanceState: Bundle?){
        binding.tvAppraise.setOnClickListener{
            context?.goShop()
            dismiss()
        }
        binding.clBg.setBackgroundResource(R.color.transparent)
        binding.tvCancel.setOnClickListener { dismiss() }
    }
}